### ISSUE TITLE
Aspects can't call `getToolsLabel()`

### ISSUE THREAD

**Issue by cgrushko** [bazelbuild/bazel#2043]
Rules can call `env.getToolsLabel("//tools/foo")` to get a label that works when there's a tools-repository, but Aspects don't have access to `env`.

This prevents Aspects from using such tools.

**Comment by cgrushko** [comment]
Thought - it would be trivial to do if we had Dagger in rules.
Just inject a string called "toolsRepository" and prepend it to labels.

**Comment by kchodorow** [comment]
Assigning to @lfpino, who removed the constant and implemented getToolsLabel.

**Comment by dslomov** [comment]
sorry what's `env`? Any documentation on this?

**Comment by kchodorow** [comment]
Looks like a [RuleDefinitionEnvironment](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/analysis/RuleDefinitionEnvironment.java), which is implemented by the ConfiguredRuleClassProvider.Builder.

**Comment by dslomov** [comment]
Oh, ok so it is not actually Skylark? It is just native rules, [`RuleDefinition.build(..)`](https://github.com/bazelbuild/bazel/blob/c484f19a2cf7427887d6e4c71c8534806e1ba83e/src/main/java/com/google/devtools/build/lib/analysis/RuleDefinition.java#L4) method?

**Comment by dslomov** [comment]
See [AndroidStudioInfoAspect constructor](https://github.com/bazelbuild/bazel/blob/124c16c8129bcdfa0fb45703d9504a34f23bd140/src/main/java/com/google/devtools/build/lib/ideinfo/AndroidStudioInfoAspect.java#L104) for how it is supposed to work.

tl;dr:Pass all the data you need to the constructor of native aspect
